client/src/Features/AssignmentLinks/AssignmentLinkEditor.tsx

```diff
@@ -34,12 +52,15 @@ const AssignmentLinkEditorInner = ({
         value={link?.displayText || ''}
         styles={themedClassNames(textFieldBaseStyle)}
         onChange={(_e, newValue) => onChangeTextField('displayText', newValue || '')}
+        onGetErrorMessage={(value) => LinkTextValidation(value, 500)}
       />
       <TextField
         value={link?.url || ''}
         placeholder="URL"
         styles={themedClassNames(urlInputStyle)}
         onChange={(_e, newValue) => onChangeTextField('url', newValue || '')}
+        onGetErrorMessage={(value) => isValidURL(value)? '' : 'Invalid URL'}

```

Comment:
I'm a bit confused as to why we've returned string for `TextLengthValidation` and not for `UrlValidation`? We could consider following the same practice for both cases. You can wrap `isValidUrl` inside the function to help this transition.